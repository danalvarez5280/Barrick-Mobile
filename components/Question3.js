import React, { Component } from 'react';
import {
  AppRegistry,
  Button,
  Image,
  Keyboard,
  Slider,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View,
} from 'react-native';

import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import concern1 from './assets/injury1.png';
import concern2 from './assets/injury2.png';
import concern3 from './assets/injury3.png';
import concern4 from './assets/injury4.png';
import concern5 from './assets/injury5.png';
import concern6 from './assets/injury6.png';


//colors
// gold #C4900F
// blue #122732
//off white #CED5E0
// off blue #324A5E
class Question3 extends Component {
  constructor(props) {
    super();
    this.state = {
      potentialInjuries: props.info.potentialInjuries,
      specificConcerns: props.info.specificConcerns
    }
    this.howConcerned = this.howConcerned.bind(this);
    this.whatConcernLevel = this.whatConcernLevel.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.goBack = this.goBack.bind(this);
  };

  goBack() {
    this.props.goBack(2);
  };

  handleSubmit() {
    console.log('specificConcerns', this.state);
    this.props.submitLogin(this.state)
  };

  noInfo() {
    Alert.alert('You need to input a SPECIFIC TASK')
  };

  howConcerned() {
    if(this.state.potentialInjuries === 1) {
      return concern1
    }
    else if(this.state.potentialInjuries === 2) {
      return concern2
    }
    else if(this.state.potentialInjuries === 3) {
      return concern3
    }
    else if(this.state.potentialInjuries === 4) {
      return concern4
    }
    else if(this.state.potentialInjuries === 5) {
      return concern5
    }
    else {
      return concern6
    }
  };

  whatConcernLevel() {
    if(this.state.potentialInjuries === 1) {
      return 'Minor Injuries'
    }
    else if(this.state.potentialInjuries === 2) {
      return 'Recordable Injuries'
    }
    else if(this.state.potentialInjuries === 3) {
      return 'Reverisble Injuries'
    }
    else if(this.state.potentialInjuries === 4) {
      return 'Irreverisble Injuries'
    }
    else if(this.state.potentialInjuries === 5) {
      return 'Single Fatality'
    }
    else {
      return 'Many Fatalities'
    }
  };

  render() {
    let concern = this.howConcerned();
    let concernLevel = this.whatConcernLevel();

    return(
      <KeyboardAwareScrollView
        scrollEnabled={ true }
        contentContainerStyle={ styles.display }>
        <View style={styles.question}>
          <Text style={ styles.prompt }>CONCERNS</Text>
          <Image source={ concern } style={styles.concernIcon}/>
          <Slider
            maximumValue={ 6 }
            maximumTrackTintColor= 'green'
            minimumValue={ 1 }
            minimumTrackTintColor= 'red'
            onSlidingComplete ={ () => this.howConcerned() }
            onValueChange={ (potentialInjuries) => this.setState({ potentialInjuries }) }
            step={ 1 }
            style={ styles.slider}
            value={ this.state.potentialInjuries }  />
            <Text>{ concernLevel }</Text>
          <Text>{ this.state.potentialInjuries }</Text>
          <Text style={ styles.prompt }>Specific Concerns</Text>
          <TextInput
            style={ styles.input }
            multiline= { true }
            placeholder='Specific Concerns'
            onChangeText={ (specificConcerns) => this.setState({ specificConcerns }) }
            value={ this.state.specificConcerns } />
          <Button onPress={ this.handleSubmit } title='SUBMIT'/>
          <Button onPress={ this.goBack } title='BACK'/>
        </View>
      </KeyboardAwareScrollView>
    )
  }
};
const styles = StyleSheet.create({
  display: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop: 10,
    padding: 0,
    height: '100%',
    width: '100%',
  },
  input: {
    backgroundColor: '#CED5E0',
    color: '#122732',
    fontSize: 20,
    height: 100,
    marginBottom: 10,
    padding: 5,
    width: 250,
  },
  question: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    marginTop: 0,
  },
  prompt: {
    backgroundColor: '#C4900F',
    color: '#122732',
    fontWeight: 'bold',
    padding: 5,
    textAlign: 'center',
    width: 250,
  },
  concernIcon: {
    height: 80,
    margin: 15,
    width: 80,
  },
  slider: {
    width: 200
  }
});

export default Question3;
