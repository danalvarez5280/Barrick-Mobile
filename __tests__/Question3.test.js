import 'react-native';
import React from 'react';
import { shallow, mount } from 'enzyme';
import Question3 from '../components/Question3';
import { Button, Image, Slider, Text, TextInput, TouchableHighlight } from 'react-native';

import renderer from 'react-test-renderer';

const q3Submit = jest.fn();
const setConcerns = jest.fn();
const logOut = jest.fn();
const goBack = jest.fn();
const mockState = {
  potentialInjuries: 1,
  specificConcerns: 'I could stub my toe',
  whatConcern: 'Minor Injuries'
};

const setup = () => {
  return shallow(<Question3
    submitLogin={ q3Submit }
    setConcerns={ setConcerns }
    goBack={ goBack }
    info={ mockState }
    logOut={ logOut } />)
};


it('renders correctly', () => {

  const tree = renderer.create(
    <Question3
    submitLogin={ q3Submit }
    goBack={ goBack }
    setConcerns={ setConcerns }
    info={ mockState }
    logOut={ logOut } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render 4 texts on the page', () => {
  let question3 = setup();

  expect(question3.find(Text).length).toEqual(4);
});

it('should render 1 image based on concern', () => {
  let question3 = setup();

  expect(question3.find(Image).length).toEqual(1);
});

it('should render 1 slider on the page', () => {
  let question3 = setup();

  expect(question3.find(Slider).length).toEqual(1);
});

it('should render 1 text input on the page', () => {
  let question3 = setup();

  expect(question3.find(TextInput).length).toEqual(1);
});

it('should render 3 buttons on the page', () => {
  let question3 = setup();

  expect(question3.find(Button).length).toEqual(3);
});

it('should be able to submit information', () => {
  let question3 = setup();

  question3.find(Button).first().simulate('press');
  expect(q3Submit).toHaveBeenCalled();
  expect(setConcerns).toHaveBeenCalled();
});

it('should be able to go back to a previous page', () => {
  let question3 = setup();

  question3.find(Button).at(1).simulate('press');
  expect(goBack).toHaveBeenCalled();
});

it('should be able to log out', () => {
  let question3 = setup();

  question3.find(Button).last().simulate('press');
  expect(logOut).toHaveBeenCalled();
});
