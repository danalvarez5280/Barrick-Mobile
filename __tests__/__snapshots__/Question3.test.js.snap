// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "height": 480,
      "justifyContent": "space-between",
      "padding": 0,
      "width": "100%",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  scrollEnabled={true}
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginTop": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        CONCERNS
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../components/assets/injury1.png",
          }
        }
        style={
          Object {
            "height": 80,
            "margin": 15,
            "width": 80,
          }
        }
      />
      <RCTSlider
        disabled={false}
        enabled={true}
        maximumTrackTintColor="green"
        maximumValue={6}
        minimumTrackTintColor="red"
        minimumValue={1}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onSlidingComplete={[Function]}
        onStartShouldSetResponder={[Function]}
        onValueChange={[Function]}
        step={1}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "width": 200,
            },
          ]
        }
        value={1}
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "#C4900F",
            "color": "#122732",
            "fontWeight": "bold",
            "padding": 5,
            "textAlign": "center",
            "width": 250,
          }
        }
      >
        Specific Concerns
      </Text>
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Specific Concerns"
        style={
          Object {
            "backgroundColor": "#CED5E0",
            "color": "#122732",
            "fontSize": 20,
            "height": 100,
            "marginBottom": 10,
            "padding": 5,
            "width": 250,
          }
        }
        value="I could stub my toe"
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            SUBMIT
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            BACK
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            LOG OUT
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
